source/language/types.rst

```diff
@@ -17,6 +17,7 @@ override a reserved identifier.
 In addition to being assigned values within a program, all of the classical
 types can be initialized on declaration. Any classical variable or Boolean that is not explicitly
 initialized is undefined. Classical types can be mutually cast to one another using the typename.
+See :ref:`castingSpecifics` for more details on casting.
```

Comment:
Maybe I missed this, but is there a specific casting syntax? Or do you initialize a new variable with that type?

Similarly, do casts happen automatically when the wrong type is returned from a function?

Reply:
Casting syntax is listed in the current spec, see the example at the end of [Integers](https://qiskit.github.io/openqasm/language/types.html#integers)
It looks like:
```
type1 identifier1 = initialization;
type2 identifier2 = type2(identifier1);
```

The `type()` syntax is the casting syntax

Reply:
> Similarly, do casts happen automatically when the wrong type is returned from a function?

This counts as implicit casting, and we are working on defining rules for that now. It is explicitly disallowed in the current spec as far as I understand.